pkg/operator/injector/injector.go

```diff
@@ -293,6 +298,29 @@ func (s *SidecarInjectField) OverlayAgent(a Annotations, ao *AnnotationOverlay,
 	return true
 }
 
+// OverlayOptional overlays optional plugins and move optional plugins to the directory(/plugins)
+// user must ensure that the optional plugins are in the injected container's image
+// Notice , user must specify the optional plugins' version
+// such as spring-cloud-gateway-2.0.x or spring-cloud-gateway-2.1.x
+// the final command will be "mv /optional-plugins/*spring-cloud-gateway-2.0.x*  /plugins/"
```

Comment:
Is "cp" here? and Do you check java agent image contains "cp" command? @kezhenxu94 Do we use this command safely in the java agent image?

Reply:
It's safe to use `cp` command in the Java agent image.

The optional plugins reminds me that we have this config `plugin.exclude_plugins=${SW_EXCLUDE_PLUGINS:}`, and as a Docker images, I'm thinking to move all optional plugins into the default `/plugins` and set `SW_EXCLUDE_PLUGINS` to the optional plugins to disable them, so that we don't need to `cp` / `mv` those files, this is also the original reason why we bring the config

Reply:
@wu-sheng mentioned that putting all optional plugins into `/plugins` will cause class loading failures. Do we fix it right now?